src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -1038,8 +1057,35 @@ protected override void WriteEntityReferenceInNavigationLinkContent(ODataNestedR
                 navigationLinkScope.EntityReferenceLinkWritten = true;
             }
 
-            Debug.Assert(entityReferenceLink.Url != null, "The entity reference link Url should have been validated by now.");
-            this.jsonWriter.WriteValue(this.jsonLightResourceSerializer.UriToString(entityReferenceLink.Url));
+            if (!this.jsonLightOutputContext.WritingResponse &&
+                (this.Version == null || this.Version < ODataVersion.V401))
+            {
+                Debug.Assert(entityReferenceLink.Url != null, "The entity reference link Url should have been validated by now.");
+                this.jsonWriter.WriteValue(this.jsonLightResourceSerializer.UriToString(entityReferenceLink.Url));

```

Comment:
should this assert be moved out of the conditional? don't we assume the entityReferenceLink.Url is set regardless of format?